test/sequence/bulk_data/discovery_sequence_test.rb

```diff
@@ -0,0 +1,172 @@
+      end
+    end
+
+    it 'fails when response header is not application/json' do
+      stub_request(:get, @well_known_endpoint)
+        .to_return(
+          status: 200,
+          headers: { content_type: 'application/text' }
+        )
+
+      assert_raises(Inferno::AssertionException) do
+        @sequence.run_test(@test)
+      end
+    end
+
+    it 'fails when response body is not JSON object' do
+      stub_request(:get, @well_known_endpoint)
+        .to_return(
+          status: 200,
+          headers: { content_type: 'application/json' },
+          body: 'This is a string'
+        )
+
+      assert_raises(Inferno::AssertionException) do
+        @sequence.run_test(@test)
+      end
+    end
+  end
+
+  describe 'well-known configuration test' do
+    before do
+      @test = @sequence_class[:validate_well_known_configuration]
+      @sequence = @sequence_class.new(@instance, @client)
+      @sequence.well_known_configuration = @smart_configuration.clone
+    end
+
+    it 'succeeds when well-known configuration is valid' do
+      @sequence.run_test(@test)
+    end
+
+    it 'fails when well-known configuration does not have required fields' do
+      @sequence.well_known_configuration.delete('token_endpoint')
+
+      assert_raises(Inferno::AssertionException) do
+        @sequence.run_test(@test)
+      end
+    end
+
+    it 'skips when well-known configuration is empty' do
+      @sequence.well_known_configuration = nil
+
+      assert_raises(Inferno::SkipException) do
+        @sequence.run_test(@test)
+      end
+    end
+  end
+
+  describe 'conformance oauth endpoints test' do
+    before do
+      @test = @sequence_class[:read_conformance_oauth_endpoins]
+      @sequence = @sequence_class.new(@instance, @client)
+    end
+
+    it 'succeeds when conformance has required oauth endpoints' do
+      stub_request(:get, @conformance_endpoint)
+        .to_return(
+          status: 200,
+          body: @conformance.to_json
+        )
+
+      @sequence.run_test(@test)
+      assert @instance.oauth_token_endpoint == 'https://bulk-data.smarthealthit.org/auth/token'
+    end
+
+    it 'fails when conformance does not have oauth extension' do
+      invalid_conformance = @conformance.clone
+      invalid_conformance['rest'][0]['security']['extension'] = nil
+      stub_request(:get, @conformance_endpoint)
+        .to_return(
+          status: 200,
+          body: @conformance.to_json
+        )
+
+      assert_raises(Inferno::AssertionException) do
+        @sequence.run_test(@test)
+      end
+    end
+
+    it 'fails when conformance does not have token endpoint' do
+      invalid_conformance = @conformance.clone
+      invalid_conformance['rest'][0]['security']['extension'][0]['extension'].shift
+      stub_request(:get, @conformance_endpoint)
+        .to_return(
+          status: 200,
+          body: @conformance.to_json
+        )
+
+      assert_raises(Inferno::AssertionException) do
```

Comment:
I think you should add an assertion to check the message on these errors. I'm assuming that this is intended to test the assertion in line 109, but the assertions in lines 109, 110, 122, and 123 look like they're not failing in any tests.

Reply:
This unit test tests for line 109 in sequence.
Next unit test tests line 110 in sequence.
I have removed line assertion in line 122 and 123 since those are not mentioned in Bulk Data